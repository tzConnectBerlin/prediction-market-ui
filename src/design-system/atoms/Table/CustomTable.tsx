import React from 'react';
import { Table, TableBody, TableCell, TableHead, TableRow } from '@material-ui/core';

export interface CustomTableProps {
  dataList: any[];
}

export const CustomTable: React.FC<CustomTableProps> = ({ dataList, ...props }) => {
  const keysList = Object.keys(dataList[0]);
  return (
    <Table aria-label="simple table">
      <TableHead>
        <TableRow>
          {keysList.map((item) => (
            <TableCell key={item}>{item}</TableCell>
          ))}
        </TableRow>
      </TableHead>
      <TableBody>
        {dataList.map((row, i) => (
          <TableRow key={i}>
            {keysList.map((item, index) => (
              <TableCell key={index}>{row[item]}</TableCell>
            ))}
          </TableRow>
        ))}
      </TableBody>
    </Table>
  );
};
