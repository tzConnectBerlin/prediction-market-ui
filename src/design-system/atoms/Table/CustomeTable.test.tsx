import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { CustomTable } from './CustomTable';

const list = [
  {
    date: '5 MAR 2021',
    adsress: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
    outcome: 'no',
    quantity: 200,
  },
  {
    date: '4 MAR 2021',
    adsress: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
    outcome: 'yes',
    quantity: 20,
  },
  {
    date: '3 MAR 2021',
    adsress: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
    outcome: 'no',
    quantity: 50,
  },
  {
    date: '2 MAR 2021',
    adsress: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
    outcome: 'yes',
    quantity: 100,
  },
];

describe('Snapshot testing Table Component', () => {
  it('renders correctly', () => {
    const Table = renderer.create(<CustomTable dataList={list} />).toJSON();
    expect(Table).toMatchSnapshot();
  });
});

describe('Element testing Table Component', () => {
  it('render correctly with mock data', async () => {
    const { container, getByText } = render(<CustomTable dataList={list} />);
    const cols = container.querySelectorAll('th');
    expect(cols.length).toBe(4);
    expect(getByText(/5 MAR 2021/i)).toBeInTheDocument();
  });
});
