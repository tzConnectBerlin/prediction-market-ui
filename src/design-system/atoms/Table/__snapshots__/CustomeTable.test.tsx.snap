// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Table Component renders correctly 1`] = `
.emotion-0 {
  display: table;
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;
}

.emotion-0 caption {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  padding: 16px;
  color: rgba(0, 0, 0, 0.6);
  text-align: left;
  caption-side: bottom;
}

.emotion-1 {
  display: table-header-group;
}

.emotion-2 {
  color: inherit;
  display: table-row;
  vertical-align: middle;
  outline: 0;
}

.emotion-2.MuiTableRow-hover:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

.emotion-2.Mui-selected {
  background-color: rgba(63, 81, 181, 0.08);
}

.emotion-2.Mui-selected:hover {
  background-color: rgba(63, 81, 181, 0.12);
}

.emotion-3 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.5rem;
  letter-spacing: 0.01071em;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 16px;
  color: rgba(0, 0, 0, 0.87);
}

.emotion-7 {
  display: table-row-group;
}

.emotion-9 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  display: table-cell;
  vertical-align: inherit;
  border-bottom: 1px solid rgba(224, 224, 224, 1);
  text-align: left;
  padding: 16px;
  color: rgba(0, 0, 0, 0.87);
}

<table
  aria-label="simple table"
  className="MuiTable-root emotion-0"
  role={null}
>
  <thead
    className="MuiTableHead-root emotion-1"
    role={null}
  >
    <tr
      className="MuiTableRow-root MuiTableRow-head emotion-2"
      role={null}
    >
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium emotion-3"
        scope="col"
      >
        date
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium emotion-3"
        scope="col"
      >
        adsress
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium emotion-3"
        scope="col"
      >
        outcome
      </th>
      <th
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium emotion-3"
        scope="col"
      >
        quantity
      </th>
    </tr>
  </thead>
  <tbody
    className="MuiTableBody-root emotion-7"
    role={null}
  >
    <tr
      className="MuiTableRow-root emotion-2"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        5 MAR 2021
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        no
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        200
      </td>
    </tr>
    <tr
      className="MuiTableRow-root emotion-2"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        4 MAR 2021
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        yes
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        20
      </td>
    </tr>
    <tr
      className="MuiTableRow-root emotion-2"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        3 MAR 2021
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        no
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        50
      </td>
    </tr>
    <tr
      className="MuiTableRow-root emotion-2"
      role={null}
    >
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        2 MAR 2021
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        yes
      </td>
      <td
        aria-sort={null}
        className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium emotion-9"
      >
        100
      </td>
    </tr>
  </tbody>
</table>
`;
